controllers/warehouse-controller.js

```diff
@@ -79,8 +79,41 @@ const deleteWarehouseById = async (req, res) => {
   }
 };
 
+//Get all inventories of a specific warehouse id (ticket 27)
+const getInventoriesOfWarehouseById = async (req, res) => {
+  try {
+    const posts = await knex("warehouses")
+      .join("inventories", "inventories.warehouse_id", "warehouses.id")
+      .where({ warehouse_id: req.params.id });
+
+    const warehousehInventoriesData = posts.map((item) => {
+      const { id, item_name, category, status, quantity } = item;
+
+      return {
+        id,
+        item_name,
+        category,
+        status,
+        quantity,
+      };
+    });
+    if (posts.length === 0) {
+      return res.status(404).json({
+        message: `Warehouse with ID ${req.params.id} is not found`,
+      });
+    } else {
+      res.status(200).json(warehousehInventoriesData);
+    }
+  } catch (error) {
+    res.status(404).json({
```

Comment:
This should be `500` since `404` is specifically for dealing with not found. `500` is for unforseen serverside errors which is what `catch` block is all about